cjworkbench/i18n/trans.py

```diff
@@ -197,13 +227,23 @@ def _format_message(self, message, parameters={}, html_escape=True):
             ],
         )
 
-    def get_message(self, message_id, context=None):
+    def get_message(
+        self, message_id: str, context: Optional[str] = None
+    ) -> Optional[str]:
         """Find the message corresponding to the given ID in the catalog.
         
         If the message is not found, `None` is returned.
         """
-        if context:
-            message = self.catalog.get(message_id, context)
-        else:
-            message = self.catalog.get(message_id)
-        return message.string if message else None
+        return find_string(self.catalog, message_id, context=context)
+
+
+def _get_translations(locale: str) -> MessageTranslator:
+    """Return a singleton MessageTranslator for the given locale.
+    
+    In order to parse the message catalogs only once per locale,
+    uses the _translators dict to store the created MessageTranslator for each locale.
+    """
+    if locale in _translators:
+        return _translators[locale]
+    _translators[locale] = MessageTranslator.for_application_messages(locale)
```

Comment:
This looks like it should be `_get_translations = lru_cache()(MessageTranslator.for_application_messages)`. Should it? If not, please comment explaining why.

(You can keep the docstring by placing it on the line after the assignment.)